src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectIdentifier.groovy

```diff
@@ -33,4 +33,17 @@ class ProjectIdentifier {
     String toString() {
         return "/${projectSpace.toString()}/${projectCode.toString()}"
     }
+
+    @Override
+    boolean equals(Object obj) {
+        if (this.is(obj)) {
+            return true
+        }
+        if (!obj instanceof ProjectIdentifier) {
+            return false
+        }
+        ProjectIdentifier otherIdentifier = (ProjectIdentifier) obj
+        return this.projectSpace.equals(otherIdentifier.getProjectSpace()) \
+                && this.projectCode.equals(otherIdentifier.getProjectCode())
```

Comment:
You could circumvent the casting here.
```suggestion
        } else {
            return this.projectSpace.equals(obj.getProjectSpace()) \
                && this.projectCode.equals(obj.getProjectCode())
        }
```
In this case `obj` is always instance of `ProjectIdentifier`

Reply:
true, I guess I just prefer the explicit style of type conversion ;)